bridge/engine/EngineMixer.cpp

```diff
@@ -2624,8 +2625,54 @@ void EngineMixer::processIncomingTransportFbRtcpPacket(const transport::RtcTrans
         return;
     }
 
+    const auto fromBarbell = EngineBarbell::isFromBarbell(transport->getTag());
     const auto mediaSsrc = rtcpFeedback->mediaSsrc.get();
 
+    if (fromBarbell)
+    {
+        const auto bb = _engineBarbells.getItem(transport->getEndpointIdHash());
+        if (bb)
+        {
+            uint32_t feedbackSsrc = 0;
+            _activeMediaList->getFeedbackSsrc(mediaSsrc, feedbackSsrc);
+
+            auto& bbTransport = bb->transport;
+            auto* mediaSsrcOutboundContext =
+                obtainOutboundSsrcContext(bb->idHash, bb->ssrcOutboundContexts, mediaSsrc, bb->audioRtpMap);
+            if (!mediaSsrcOutboundContext || !mediaSsrcOutboundContext->packetCache.isSet() ||
+                !mediaSsrcOutboundContext->packetCache.get())
+            {
+                return;
+            }
+
+            auto* feedbackSsrcOutboundContext =
+                obtainOutboundSsrcContext(bb->idHash, bb->ssrcOutboundContexts, feedbackSsrc, bb->audioRtpMap);
+            if (!feedbackSsrcOutboundContext)
+            {
+                return;
+            }
+
+            mediaSsrcOutboundContext->onRtpSent(timestamp);
+            const auto numFeedbackControlInfos = rtp::getNumFeedbackControlInfos(rtcpFeedback);
+            uint16_t pid = 0;
+            uint16_t blp = 0;
+            for (size_t i = 0; i < numFeedbackControlInfos; ++i)
+            {
+                feedbackSsrcOutboundContext->onRtpSent(timestamp);
+                rtp::getFeedbackControlInfo(rtcpFeedback, i, numFeedbackControlInfos, pid, blp);
+                bbTransport.getJobQueue().addJob<bridge::VideoNackReceiveJob>(*feedbackSsrcOutboundContext,
+                    bbTransport,
+                    *(mediaSsrcOutboundContext->packetCache.get()),
+                    pid,
+                    blp,
+                    feedbackSsrc,
+                    timestamp,
+                    transport->getRtt());
+            }
+        }
+        return;
```

Comment:
This is quite extensive method and in the end has an early return. I would prefer having the early returns not far way from the condition. Could you put this in other method. like:
```
if (fromBarbell)
{
    processIncomingFbRtcpPacketFromBarbellTransport(transport, rtcpPacket, timestamp);
    return;
}